Trial balance and payable accounts reports and sales return fixes

elba7r opened the pull request:
…summary reports

adding supplier type filter to payable accounts and payable accounts
summary reports as customer group at receivable accounts and summary
reports as its important and easier to put this filter
also  there is a fix for the opening and closing values at the trial balance report as the deleted operation was causing incorrect values and not equal to the total row values at opening and closing so now its correct and equal to the total row values

Please read the pull request checklist to make sure your changes are merged: https://github.com/frappe/erpnext/wiki/Pull-Request-Checklist

elba7r committed:
adding supplier type filter to payable accounts and payable accounts summary reports

adding supplier type filter to payable accounts and payable accounts
summary reports as customer group at receivable accounts and summary
reports as its important and easier to put this filter

elba7r committed:
fix trial balance opening and closing values

the removed mathematical operations was causing showing incorrect values
at opening and closing columns where the total row was correct so the
operation deleted to view the correct values to be equal to the total
row and to stop confusing the user due to different values between the
total row and the the values in front of each account

nabinhait commented on erpnext/accounts/report/accounts_payable/accounts_payable.js:
```diff
@@ -28,6 +28,12 @@ frappe.query_reports["Accounts Payable"] = {
+			"fieldname":"supplier_type",
+			"label": __("Supplier Type"),
+			"fieldtype": "Link",
+			"options": "Supplier Type"
```
"Supplier Type" has been renamed to "Supplier Group" in develop branch.

elba7r replied:
thats ok i have applied this requests

nabinhait commented on erpnext/accounts/report/accounts_receivable/accounts_receivable.py:
```diff
@@ -378,6 +378,14 @@ def prepare_conditions(self, party_type):
+   		    if self.filters.get("supplier_type"):
+		        suptype = self.filters.get("supplier_type")
+    		        conditions.append("""party in (select name from tabSupplier
+		                where supplier_type=%s)""")
```
`supplier_type` field also renamed to `supplier_group`

elba7r replied:
thats ok i have applied this requests and going to apply the pull request again

nabinhait reviewed (changes requested):
Can you add a attachment showing the problem in trial balance?

nabinhait commented:
@elba7r Check if https://github.com/frappe/erpnext/pull/14990 solves the problem in trial balance.

elba7r commented:
dus to trial balance the applied pull request which is mentioned in your comment not fixing the problem but it makes it worse and destroy the meaning of trial balance report.
that way of showing the trial balance is giving incorrect values and cause misunderstanding for the real numbers in the accounts .also it is not fix the opening values.
my pull request is fixing the problem from the root as you should not make any operations on the opening and closing ( debit - credit ) columns and you should show the real amounts on the accounts and this is what my pull request do :)

elba7r committed:
fix for supplier type filter field on payable accounts reports

rename the supplier type filter field to supplier group in payable
accounts and summary reports

elba7r commented:
Here is the print screen for trial balance before my edits
![trial_balance_before-edits](https://user-images.githubusercontent.com/23710710/43368075-6c0879ec-9357-11e8-8a38-696c89502309.png)

here is the correct trial balance after my edits

![trial_balance_after-edits](https://user-images.githubusercontent.com/23710710/43368081-8f2db86a-9357-11e8-9542-7a6f8c4f3602.png)

elba7r committed:
fix for paid amount in returned sales invoice

the current paid amount not change from source value if you deleted any
item if the invoice contain for example 3 items and you only need to
return on item the current steps is to open the original invoice and
press on make return then you delete the unwanted items but in this case
the paid amount will not change and you will get an alert from the
system until you change it manually but with this modification it works
very well.

FinForceConsulting commented:
@elba7r - "here is the correct trial balance after my edits' 
This is grossly incorrect as per the accounting conventions. Trial balance is a summary of 'Net balance' in the ledger account. And one Ledger account can have either a Dr or a Cr balance. What is see in the 'before' picture seems fine to me as it is showing only the Net balances. But in the 'After' it shows total of all the Dr entries and total of the Cr entries, which is incorrect.

Showing total of Dr & Cr line items in the Transaction column (i.e. column 3 & 4) is fine, but for the opening Balances and for the closing balances, it should ONLY show one balance, either Dr or Cr, as applicable appropriately.

"that way of showing the trial balance is giving incorrect values and cause misunderstanding for the real numbers in the accounts .also it is not fix the opening values."

The #14990 fixes the problem of Sum, that is good. But your opening balance showing total of Dr separately and Cr separately is giving no meaning. That is not the way an Accountant looks at the report.

@nabinhait, Unless we have a strong Use Case for this PR request of advantanges of showing totals of Dr & Cr separataly in the Opening & Closing balances, this PR should not be merged. Thanks.

Regards
CA Liyakat

elba7r commented:
@FinForceConsulting  
here is a print screen for the current trial balance without my changes
![printscreen-showing the current error](https://user-images.githubusercontent.com/23710710/43497269-9c8f908c-9541-11e8-81ec-44cb20d54764.png)

i believe showing the full values CR and DR is much better than showing the report in this way as if you focus you will see that total row is not correct if you filtered and period during the year.

please remove the do not merge flag until we find a better way showing the trial balance with correct totals in your way

elba7r committed:
fixing trial balance values and totals

fixing trial balance opening and closing ( DR - CR ) values and totals

elba7r commented:
@FinForceConsulting  @nabinhait 

Now i fixed the trial balance at the correct way and the values and totals are correct and here is a print screen after correction

![correct-trial-balance](https://user-images.githubusercontent.com/23710710/43645395-032b7cce-9732-11e8-85eb-e694b238df65.png)

elba7r committed:
Merge branch 'develop' into develop

elba7r reviewed (commented):
i have applied all required requests

FinForceConsulting commented:
@elba7r, Great! So, how **different** your contribution will make the Trial Balance from the existing Trial Balance? Sorry! I'm not a coder, so you have to explain to me the functional side of it.

@nabinhait, Let's discuss anything from the accounting side before you may decide to take this into the merge. Thanks.

_Liyakat

elba7r commented:
@FinForceConsulting @nabinhait 
The current trial balance opening and closing was looking for which field is bigger if dr > cr so dr = dr-cr and so on and this was causing some values does not appear in totals and cause confusion, but my change making this operation depends on the root type of the account i put screen shots for the same period for the same company before and after the edits which may explain the changes more clear : 

Here is the print screen for trial balance before my changes : 

![current-trial-balance-1](https://user-images.githubusercontent.com/23710710/43675529-233b3e04-97e2-11e8-904e-790b6deb46f7.png)

![current-trial-balance-2](https://user-images.githubusercontent.com/23710710/43675532-27a51078-97e2-11e8-843c-909bea6ad34d.png)

![current-trial-balance-3](https://user-images.githubusercontent.com/23710710/43675535-2c2ca7d2-97e2-11e8-86e3-11c1b23b9edc.png)

![current-trial-balance-4](https://user-images.githubusercontent.com/23710710/43675538-2fa4c28c-97e2-11e8-91d0-63c256c07756.png)

![current-totals](https://user-images.githubusercontent.com/23710710/43675543-365fa02e-97e2-11e8-8668-390fe54c7f48.png)


Here is the same period and company after fixing : 

![fixed-trial-balance-1](https://user-images.githubusercontent.com/23710710/43675548-444d80c0-97e2-11e8-8ff3-312f5566429e.png)

![fixed-trial-balance-2](https://user-images.githubusercontent.com/23710710/43675550-4a1affa0-97e2-11e8-9a57-d34544248c45.png)

![fixed-trial-balance-3](https://user-images.githubusercontent.com/23710710/43675554-4e3e0d2a-97e2-11e8-87ba-f8e6c650065b.png)

![fixed-trial-balance-4](https://user-images.githubusercontent.com/23710710/43675555-52de9106-97e2-11e8-825b-5e059ee47c55.png)

![fixed-totals](https://user-images.githubusercontent.com/23710710/43675557-56c0e832-97e2-11e8-8421-bf790ca4961f.png)

Best Regards

elba7r commented:
@nabinhait  @FinForceConsulting 
Today The stale put a flag inactive and the pull request will be closed automatically, so please any news about this pull request as i still have important feature for the pos requested from the community ( cashier closing )  i wants to push, and also a fix for the pos duplicates invoices, but i'm waiting until this pull request to be merged.
So please inform me !

Best Regards
